mlflow/server/js/src/components/MetricsPlotPanel.js

```diff
@@ -106,13 +111,117 @@ export class MetricsPlotPanel extends React.Component {
+    // for log axis (base 10), and vice versa. When converting to log scale, handle negative values
+    // by deferring to plotly autorange
+    const newAxisType = yAxisLogScale ? "log" : "linear";
+
+    // If plot previously had no y axis range configured, simply set the axis type to log or
+    // linear scale appropriately
+    if (!this.state.layout.yaxis || !this.state.layout.yaxis.range) {
+      newLayout.yaxis = {type: newAxisType, autorange: true};
+      this.setState({layout: newLayout});
+      return;
+    }
+
+    // At this point, we know the plot previously had a y axis specified with range
+    // Convert the range to/from log scale as appropriate
+    const oldLayout = this.state.layout;
+    const oldYRange = oldLayout.yaxis.range;
+    if (oldLayout.yaxis.type !== 'log') {
+      if (oldYRange[0] < 0) {
+        // When converting to log scale, handle negative values as follows:
+        // If bottom of old Y range is negative, then tell plotly to infer the y-axis scale
+        // (set 'autorange' to true), and preserve the old range in the 'range' attribute of
+        // the layout so that we can restore it if the user converts back to a linear-scale
+        // y axis.
+        newLayout.yaxis = {
+          type: 'log',
+          range: oldYRange,
+          autorange: true,
+        };
+      } else {
+        newLayout.yaxis = {
+          type: 'log',
+          range: [Math.log(oldYRange[0]) / Math.log(10), Math.log(oldYRange[1]) / Math.log(10)],
+        };
+      }
+    // Handle conversion from log to linear scale. If old Y axis was autoranged (had a range
+    // inferred by Plotly), this means that there were negative Y values (on a linear scale).
+    // We handle this by removing the 'autorange' value and restoring the old y range
+    } else if (oldLayout.yaxis.autorange) {
+      newLayout.yaxis = {
+        type: 'linear',
+        range: oldYRange,
+      };
+    } else {
+      // Otherwise, convert from log to linear scale normally
+      newLayout.yaxis = {
+        type: 'linear',
+        range: [Math.pow(10, oldYRange[0]), Math.pow(10, oldYRange[1])],
+      };
+    }
+    this.setState({layout: newLayout});
   };
 
   handleXAxisChange = (e) => {
-    this.setState({ selectedXAxis: e.target.value });
+    // Set axis value type, & reset axis scaling via autorange
+    const axisType = e.target.value === X_AXIS_WALL ? "date" : "linear";
+    const newLayout = {
+      ...this.state.layout,
+      xaxis: {
+        autorange: true,
+        type: axisType,
+      },
+    };
+    this.setState({ selectedXAxis: e.target.value, layout: newLayout });
+  };
+
+  handleLayoutChange = (newLayout) => {
+    // Unfortunately, we need to parse out the x & y axis range changes from the onLayout event...
+    // see https://plot.ly/javascript/plotlyjs-events/#update-data
+    const newXRange0 = newLayout["xaxis.range[0]"];
```

Comment:
Does it have to be accessed using a path? If not we can simplify this as 
```
const [newXRangeStart, newXRangeEnd] = newLayout.xaxis;
```

Reply:
Unfortunately it does - the actual key is called `"xaxis.range[0]"`